### ISSUE TITLE
[Feature]: vectorise the decode mask tiling in build()

### ISSUE THREAD

**Issue by jvlunteren** [torch-spyre/spyre-inference#786]
### Feature description

`SpyreAttentionMetadataBuilder.build()` splits the attention mask into per-block tiles with one slice per (sequence, block):

```
for s in range(num_seqs):
    seq_tiles: list[torch.Tensor] = []
    for b in range(padded_num_blocks[s]):
        tile = mask_cpu[s, :aligned_max_query_len, b * block_size : (b + 1) * block_size]
        seq_tiles.append(tile.contiguous())
    attention_mask_tiles.append(seq_tiles)
```

At decode `aligned_max_query_len` is 1, so every tile is a single row and the loop's cost is dominated by constructing `num_seqs * padded_num_blocks` tensor objects rather than by moving the data. A batched copy of the whole mask is a small fraction of what the individual constructions cost.

### Motivation and context

`build()` runs on every decode step, and vLLM calls it from _prepare_inputs before the model forward rather than alongside it, so this is host time on the critical path rather than work hidden behind device execution.

The cost scales with `num_seqs × padded_num_blocks`, which means it grows with both batch width and context length. Profiling `build()` at decode geometries showed this loop to be its dominant term, larger than `_build_attention_mask`, which builds the mask the loop then slices. So a wide batch at long context pays the most, which is the regime the bucketed decode path is otherwise most attractive for.

### Proposed solution

Replace the per-tile slicing with one batched copy followed by unbind, which hands back the same per-block tensors in a single call.

Two constraints make this narrower than it first appears, and both are correctness-relevant:

1. `_mirror_mask_tiles` dedupes by `id()`, so the tiles must remain distinct tensor objects. `unbind` satisfies this, a single fused tensor would not.
2. Storage offsets must not change. At `aligned_max_query_len == 1` a one-row slice is already contiguous, so `.contiguous()` is a no-op returning the view. Today's tiles already carry nonzero offsets and unbind reproduces them. Above 1 the slice is non-contiguous and `.contiguous()` clones to offset 0, so substituting unbind there would change what the tiles are, which matters given torch-spyre #3770.

Gating the fast path on `aligned_max_query_len == 1` keeps the change equivalent and loses nothing: prefill builds few large tiles rather than many small ones, so the per-object cost this targets is not present there.

A test comparing the vectorised construction against the original loop (asserting values, storage offsets, contiguity and tensor identity across ragged and prefill shapes) is the cheapest way to keep it honest, and needs no device.

### Checklist

- [x] I have searched for similar feature requests
